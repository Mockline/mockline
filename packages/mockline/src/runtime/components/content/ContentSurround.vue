<script setup lang="ts">
import type { PropType } from 'vue'
import type { ParsedContent } from '@nuxt/content/dist/runtime/types'

const props = defineProps({
  next: {
    type: Object as PropType<ParsedContent>,
    default: null
  },
  prev: {
    type: Object as PropType<ParsedContent>,
    default: null
  }
})
</script>

<template>
  <div class="p-4">
    <div class="flex justify-between">
      <NuxtLink v-if="prev" :to="prev._path" class="bg-canvas-3 flex items-center rounded p-2">
        <div class="flex items-center gap-2">
          <span class="i-lucide-arrow-left size-4" />
          <span>{{ prev.title }}</span>
        </div>
      </NuxtLink>
      <NuxtLink v-if="next" :to="next._path" class="bg-canvas-3 flex items-center rounded p-2">
        <div class="flex items-center gap-2">
          <span>{{ next.title }}</span>
          <span class="i-lucide-arrow-right size-4" />
        </div>
      </NuxtLink>
    </div>
  </div>
</template>
