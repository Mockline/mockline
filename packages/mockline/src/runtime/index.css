@import "tailwindcss";
@import "assets/css/animation.css";

@variant dark (&:where(.dark, .dark *));

@plugin "@tailwindcss/typography";

@theme {
  --color-neutral: var(--ui-bg);
  --color-neutral-elevated: var(--ui-bg-elevated);
  --color-border: var(--ui-border);
}

@layer base {
  :root {
    color-scheme: light;

    --ui-text-dimmed: var(--ui-color-neutral-400);
    --ui-text-muted: var(--ui-color-neutral-500);
    --ui-text-toned: var(--ui-color-neutral-600);
    --ui-text: var(--ui-color-neutral-700);
    --ui-text-highlighted: var(--ui-color-neutral-900);

    --ui-bg: var(--color-white);
    --ui-bg-elevated: var(--ui-color-neutral-100);
    --ui-bg-accented: var(--ui-color-neutral-200);
    --ui-bg-inverted: var(--ui-color-neutral-900);

    --ui-border: var(--ui-color-neutral-200);
    --ui-border-accented: var(--ui-color-neutral-300);
    --ui-border-inverted: var(--ui-color-neutral-900);

    --ui-radius: var(--radius);

    /*--ui-slate: var(--color-slate-500);
    --ui-gray: var(--color-gray-500);
    --ui-zinc: var(--color-zinc-500);
    --ui-stone: var(--color-stone-500);
    --ui-neutral: var(--color-neutral-500);
    --ui-red: var(--color-red-500);
    --ui-orange: var(--color-orange-500);
    --ui-amber: var(--color-amber-500);
    --ui-yellow: var(--color-yellow-500);
    --ui-lime: var(--color-lime-500);
    --ui-green: var(--color-green-500);
    --ui-emerald: var(--color-emerald-500);
    --ui-teal: var(--color-teal-500);
    --ui-cyan: var(--color-cyan-500);
    --ui-sky: var(--color-sky-500);
    --ui-blue: var(--color-blue-500);
    --ui-indigo: var(--color-indigo-500);
    --ui-violet: var(--color-violet-500);
    --ui-purple: var(--color-purple-500);
    --ui-fuchsia: var(--color-fuchsia-500);
    --ui-pink: var(--color-pink-500);
    --ui-rose: var(--color-rose-500);*/
  }

  .dark {
    color-scheme: dark;

    --ui-text-dimmed: var(--ui-color-neutral-500);
    --ui-text-muted: var(--ui-color-neutral-400);
    --ui-text-toned: var(--ui-color-neutral-300);
    --ui-text: var(--ui-color-neutral-200);
    --ui-text-highlighted: var(--color-white);

    --ui-bg: var(--ui-color-neutral-900);
    --ui-bg-elevated: var(--ui-color-neutral-800);
    --ui-bg-accented: var(--ui-color-neutral-700);
    --ui-bg-inverted: var(--color-white);

    --ui-border: var(--ui-color-neutral-800);
    --ui-border-accented: var(--ui-color-neutral-700);
    --ui-border-inverted: var(--color-white);
  }
}

