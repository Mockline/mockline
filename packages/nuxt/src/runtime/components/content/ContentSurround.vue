<script setup lang="ts" generic="T extends ContentSurroundLink">
import type { ContentNavigationItem } from '@nuxt/content'

export type ContentSurroundLink = {
  description?: string
  icon?: string
} & ContentNavigationItem

export type ContentSurroundProps<T> = {
  /**
   * The element or component this component should render as.
   * @defaultValue 'div'
   */
  as?: any
  /**
   * The icon displayed in the prev link.
   * @defaultValue appConfig.mockline.icons.arrowLeft
   */
  prevIcon?: string
  /**
   * The icon displayed in the next link.
   * @defaultValue appConfig.mockline.icons.arrowRight
   */
  nextIcon?: string
  surround?: T[]
  class?: any
}

const { surround } = defineProps<ContentSurroundProps<T>>()

const prev = surround?.[0]
const next = surround?.[1]
</script>

<template>
  <div class="p-4">
    <div class="flex justify-between">
      <NuxtLink v-if="prev" :to="prev.path" class="bg-canvas-3 flex items-center rounded p-2">
        <div class="flex items-center gap-2">
          <Icon name="lucide:arrow-left" class="size-4" />
          <div class="flex flex-col">
            <span>{{ prev.title }}</span>
            <div class="text-xs text-neutral-500 dark:text-neutral-400">
              {{ prev.description }}
            </div>
          </div>
        </div>
      </NuxtLink>
      <NuxtLink v-if="next" :to="next.path" class="bg-canvas-3 flex items-center rounded p-2">
        <div class="flex items-center gap-2">
          <div class="flex flex-col items-end">
            <span>{{ next.title }}</span>
            <div class="text-xs text-neutral-500 dark:text-neutral-400">
              {{ next.description }}
            </div>
          </div>
          <Icon name="lucide:arrow-right" class="size-4" />
        </div>
      </NuxtLink>
    </div>
  </div>
</template>
