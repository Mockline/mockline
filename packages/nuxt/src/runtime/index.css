@import "@mockline/themes";
@import "./mockline.css";
