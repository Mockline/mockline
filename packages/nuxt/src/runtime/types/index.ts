// Elements
export * from '../components/Button.vue'
export * from '../components/NavMenu.vue'

// Layout
export * from '../components/App.vue'
export * from '../components/Aside.vue'
export * from '../components/Card.vue'
export * from '../components/Header.vue'
export * from '../components/Main.vue'
export * from '../components/Page.vue'
export * from '../components/PageBody.vue'
export * from '../components/PageHeader.vue'

// Settings
export * from '../components/ThemeToggle.vue'

// Overlays
export * from '../components/Toasts.vue'

// Content
export * from '../components/content/ContentNavigationTree.vue'
export * from '../components/content/ContentSurround.vue'
export * from '../components/content/ContentToc.vue'
export * from '../components/content/ContentTocLinks.vue'

// Global
export * from '../components/prose/Callout.vue'
export * from '../components/prose/CodeGroup.vue'
export * from '../components/prose/ProseCode.vue'
export * from '../components/prose/CodeButton.vue'
