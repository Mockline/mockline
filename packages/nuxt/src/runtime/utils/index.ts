export function transformSlot(slot: any, index: number): any {
  if (typeof slot.type === 'symbol') {
    return slot.children?.map(transformSlot)
  }
  return {
    label: slot.props?.filename || slot.props?.label || `${index}`,
    icon: slot.props?.icon,
    component: slot
  }
}

export * from './useComponent'
