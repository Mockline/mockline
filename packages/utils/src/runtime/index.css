@import "tailwindcss";

@variant dark (&:where(.dark, .dark *));

@plugin "@tailwindcss/typography";
