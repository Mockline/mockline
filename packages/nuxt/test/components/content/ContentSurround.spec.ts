import { describe, it, expect } from 'vitest'
import { renderSuspended } from '@nuxt/test-utils/runtime'
import ContentSurround from '../../src/runtime/components/content/ContentSurround.vue'

describe('ContentSurround', () => {
  it('can mount component', async () => {
    const html = await renderSuspended(ContentSurround)
    expect(html.html()).toMatchSnapshot()
  })
})
