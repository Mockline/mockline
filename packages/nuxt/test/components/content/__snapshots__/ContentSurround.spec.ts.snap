// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContentSurround > renders <MContentSurround>ContentSurround slot</MContentSurround> correctly 1`] = `
"<div class="p-4">
  <div class="flex justify-between">
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`ContentSurround > renders basic case correctly 1`] = `
"<div class="p-4">
  <div class="flex justify-between"><a href="path" class="bg-canvas-3 flex items-center rounded p-2">
      <div class="flex items-center gap-2"><span class="iconify i-lucide:arrow-left size-4" aria-hidden="true"></span>
        <div class="flex flex-col"><span>title</span>
          <div class="text-xs text-neutral-500 dark:text-neutral-400"></div>
        </div>
      </div>
    </a>
    <!--v-if-->
  </div>
</div>"
`;
